### ISSUE TITLE
media attribute is allowed on <meta name=theme-color>

### ISSUE THREAD

**Issue by mtrootyy** [validator/validator#1166]
Ref: https://github.com/whatwg/html/commit/cf222a32f3ffba0513a3f5dde1acd2a3ede794e1

**Comment by tannerdolby** [comment]
I tried validating with `<meta name="theme-color" content="#f06">` and the validator doesn't throw any errors or warnings. 

<img width="1276" alt="Screen Shot 2021-07-10 at 2 49 36 PM" src="https://user-images.githubusercontent.com/48612525/125177111-20649380-e18e-11eb-83c3-a90ed64ba6b2.png">


I had a look at `Assertions.java`,

https://github.com/validator/validator/blob/cc73b75739f8e664e24410de4413bb625020f50c/src/nu/validator/checker/schematronequiv/Assertions.java#L2945-L2951


and I think maybe this has already been fixed here in validator @sideshowbarker ?

**Comment by sideshowbarker** [comment]
This issue is specifically about the `media` attribute being allowed with the `meta` element, like this:

```html
<meta name="theme-color" content="#3c790a" media="(prefers-color-scheme: dark)">
```

**Comment by tannerdolby** [comment]
Ah ok that makes more sense. Thanks for the heads up!

**Comment by sdavids** [comment]
https://html.spec.whatwg.org/multipage/semantics.html#meta-theme-color

> The [media](https://html.spec.whatwg.org/multipage/semantics.html#attr-meta-media) attribute may be used to describe the context in which the provided color should be used.

https://html.spec.whatwg.org/multipage/semantics.html#attr-meta-media

> The media attribute says which media the metadata applies to. The value must be a [valid media query list](https://html.spec.whatwg.org/multipage/common-microsyntaxes.html#valid-media-query-list). Unless the [name](https://html.spec.whatwg.org/multipage/semantics.html#attr-meta-name) is [theme-color](https://html.spec.whatwg.org/multipage/semantics.html#meta-theme-color), the [media](https://html.spec.whatwg.org/multipage/semantics.html#attr-meta-media) attribute has no effect on the processing model and must not be used by authors.

**Comment by Rotzbua** [comment]
Observed the same problem. 
## Testdata

```html
<!DOCTYPE html>
<html lang="en">
<head>
<title>HTML Standard</title>
<meta name="theme-color" content="#3c790a" media="(prefers-color-scheme: dark)">
</head>
</html>
```

## Error message

> Error: Attribute media not allowed on element `meta` at this point.
> From line 5, column 1; to line 5, column 80

## Expected

No error, because it is the example from the specification:
https://html.spec.whatwg.org/multipage/semantics.html#meta-theme-color

**Comment by jfrech** [comment]
I suspect the fix for this is non-trivial, as [the HTML parser itself](https://github.com/validator/htmlparser/blob/master/src/nu/validator/htmlparser/impl/MetaScanner.java) seems to reject the `media=` attribute. Probably because of this, the code section @tannerdolby linked to does not appear to produce this phenomenon.
